src/webapp/pages/settings/SettingsPage.tsx

```diff
@@ -0,0 +1,136 @@
+import i18n from "@dhis2/d2-i18n";
+import { TableAction, ObjectsTable, useSnackbar } from "@eyeseetea/d2-ui-components";
+import { Button } from "@dhis2/ui";
+import React, { useEffect, useMemo } from "react";
+import { PageHeader } from "../../components/page-header/PageHeader";
+import { useGoBack } from "../../hooks/useGoBack";
+import { ListOptions } from "../../../domain/repositories/MetadataRepository";
+import { MetadataItem } from "../../../domain/entities/MetadataItem";
+import RemoveCircleOutlineIcon from "@material-ui/icons/RemoveCircleOutline";
+import AddCircleOutlineIcon from "@material-ui/icons/AddCircleOutline";
+import { EllipsizedList } from "../../components/ellipsized-list/EllipsizedList";
+import { useSettingsPage } from "./useSettingsPage";
+import { Filter } from "./Filter";
+
+export const SettingsPage: React.FC = () => {
+    const goBack = useGoBack();
+    const snackbar = useSnackbar();
+
+    const {
+        errorMessage,
+        successMessage,
+        isLoading,
+        response,
+        selection,
+        listOptions,
+        excludeDependency,
+        getModelName,
+        includeDependency,
+        isExcludeActive,
+        isIncludeActive,
+        onChangeSearch,
+        onChangeModel,
+        onTableChange,
+        rowConfig,
+        saveExcludedDependencies,
+    } = useSettingsPage();
+
+    useEffect(() => {
+        if (!errorMessage) return;
+        else snackbar.error(errorMessage);
+    }, [errorMessage, snackbar]);
+
+    useEffect(() => {
+        if (!successMessage) return;
+        else snackbar.success(successMessage);
```

Comment:
you don't need an else because if the if is fulfilled, a return is made (also above)